test/groovy/UpdatecliStepTests.groovy

```diff
@@ -136,4 +141,42 @@ class UpdatecliStepTests extends BaseTest {
     // And the custom credentialsId is taken in account
     assertTrue(assertMethodCallContainsPattern('usernamePassword', "credentialsId=${anotherCredentialsId}"))
   }
+
+  /**
+   * New Test Cases for Custom Version Functionality
+   */
+
+  // Test that when a custom version is specified, the pipeline includes the download steps.
+  @Test
+  void itRunSuccessfullyWithCustomVersion() throws Exception {
+    def script = loadScript(scriptName)
+    script.call(version: '0.92.0')
+    printCallStack()
+    assertJobStatusSuccess()
+    assertTrue(assertMethodCallContainsPattern('sh', 'curl --silent --show-error --location --output'))
+    assertTrue(assertMethodCallContainsPattern('sh', 'mkdir -p "${CUSTOM_UPDATECLI_PATH}"'))
+    assertTrue(assertMethodCallContainsPattern('sh', 'tar --extract --gzip --file="${tarFileName}" --directory="${CUSTOM_UPDATECLI_PATH}" updatecli'))
+    assertTrue(assertMethodCallContainsPattern('sh', 'updatecli diff'))
+  }
+
+  @Test
+  void itFailsWhenCustomVersionNotFound() throws Exception {
+    helper.registerAllowedMethod('sh', [Map.class], { Map args ->
+      if (args.script.contains("curl --silent --show-error --location --output") && args.script.contains("updatecli_Linux")) {
+        return 1  // simulate download failure (non-zero exit status)
+      }
+      if (args.script.contains("uname -m")) {
+        return "x86_64\n"
+      }
+      return 0
+    })
+
+    def script = loadScript(scriptName)
+    try {
+      script.call(version: '0.99.99')
+      fail("Expected error due to custom version not found")
+    } catch (Exception e) {
+      assertTrue(assertMethodCallContainsPattern('sh', 'curl --silent --show-error'))
+    }
```

Comment:
```suggestion
    }
```

Missing an `assertFalse(assertJobStatusSuccess())` to verify the pipeline simulated executions fails